bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemRegistryImpl.java

```diff
@@ -245,12 +241,11 @@ private void addToGroupItems(Item item, List<String> groupItemNames) {
      * injected and its implementation is notified that it has just been
      * created, so it can perform any task it needs to do after its creation.
      *
-     * @param item
-     *            the item to initialize
+     * @param item the item to initialize
      * @throws IllegalArgumentException if the item has no valid name
      */
     private void initializeItem(Item item) throws IllegalArgumentException {
-        if (isValidItemName(item.getName())) {
+        if (ItemUtil.isValidItemName(item.getName())) {
```

Comment:
Here you should also use the assert, since the else branch also only throws an IAE.